CHANGELOG.md

```diff
@@ -5,6 +5,7 @@ All notable changes to this project will be documented in this file.
 
 * **BREAKING CHANGE** Use `id` as an alias for `_id` in commands and queries for compatibility with Eloquent packages by @GromNaN in [#3040](https://github.com/mongodb/laravel-mongodb/pull/3040)
 * **BREAKING CHANGE** Make Query\Builder return objects instead of array to match Laravel behavior by @GromNaN in [#3107](https://github.com/mongodb/laravel-mongodb/pull/3107)
+* **BREAKING CHANGE** Convert BSON `UTCDateTime` objects into `Carbon` date in query results by @GromNaN in [#3119](https://github.com/mongodb/laravel-mongodb/pull/3119)
```

Comment:
Would it be worth noting that the converted object gets the default timezone applied?

If Carbon objects were previously converted to UTCDateTime and left as-is, I suppose users would already be accustomed to having any original time zone dropped for UTC. But seeing that Carbon types are now returned might lead them to believe that the original Carbon object's timezone is restored, which isn't necessarily the case.

Doing so would require preserving the original timezone alongside the UTCDateTime in some kind of nested object (see POC in [PHPC-640](https://jira.mongodb.org/browse/PHPC-640)). I'm not advocating for such an implementation -- just pointing out that it was a consideration during the early designs of PHPC's BSON APIs.

Reply:
Storing the timezone is something we could consider with custom date cast (need a [mutator](https://laravel.com/docs/11.x/eloquent-mutators))